src/test/moment/getters_setters.js

```diff
@@ -120,36 +120,30 @@ test('setter programmatic', function (assert) {
     assert.equal(a.month(), 3, 'month edge case');
 });
 
-// Disable this, until we weekYear setter is fixed.
-// https://github.com/moment/moment/issues/1379
-// test('setters programatic with weeks', function (assert) {
-//     var a = moment();
-//     a.set('weekYear', 2001);
-//     a.set('week', 49);
-//     a.set('day', 4);
-//     assert.equals(a.weekYear(), 2001);
-//     assert.equals(a.week(), 49);
-//     assert.equals(a.day(), 4);
-
-//     a.set('weekday', 1);
-//     assert.equals(a.weekday(), 1);
+test('setters programatic with weeks', function (assert) {
+    var a = moment();
+    a.set('weekYear', 2001);
+    a.set('week', 49);
+    a.set('day', 4);
 
-//     assert.done();
-//},
+    assert.equal(a.weekYear(), 2001, 'weekYear');
+    assert.equal(a.week(), 49, 'week');
+    assert.equal(a.day(), 4, 'day');
 
-// I think this suffers from the same issue as the non-iso version.
-// test('setters programatic with weeks ISO', function (assert) {
-//     var a = moment();
-//     a.set('isoWeekYear', 2001);
-//     a.set('isoWeek', 49);
-//     a.set('isoWeekday', 4);
+    a.set('weekday', 1);
+    assert.equal(a.weekday(), 1, 'weekday');
+});
 
-//     assert.equals(a.weekYear(), 2001);
-//     assert.equals(a.week(), 49);
-//     assert.equals(a.day(), 4);
+test('setters programatic with weeks ISO', function (assert) {
+    var a = moment();
+    a.set('isoWeekYear', 2001);
+    a.set('isoWeek', 49);
+    a.set('isoWeekday', 4);
 
-//     assert.done();
-//},
+    assert.equal(a.weekYear(), 2001, 'weekYear');
+    assert.equal(a.week(), 49, 'week');
+    assert.equal(a.day(), 4, 'day');
+});
```

Comment:
Why are the weekYear gets being compared to isoWeekYear sets? This happens to be true when you're in the middle of the year, but consider:

```js
moment('2012-01-01').set('isoWeekYear', 2016).format()
"2017-01-01T00:00:00-06:00"
moment('2012-01-01').set('weekYear', 2016).format()
"2015-12-27T00:00:00-06:00"
```

Also, 

```js
moment('2012-01-01').set('isoWeekYear', 2016).weekYear()
2017
```
What don't I understand?